### ISSUE TITLE
"workaround" needles should predictably have precedence over regular needles when both match

### ISSUE THREAD

**Issue by AdamWill** [os-autoinst/openQA#779]
Per IRC discussion:

    <adamw> hmmm
    <adamw> https://openqa.fedoraproject.org/tests/24758#step/_graphical_wait_login/1
    <adamw> shouldn't a 100% match of a non-workaround needle beat a 100% match of a workaround needle?
    <okurz> adamw: I don't know right now which needles are accepted first but actually I prefer it this way: workaround needles used for tracking known issues can be added afterwards if an issue is seen in a "green" testcase
    <coolo> adamw, okurz: if two needles match the first in alphabetical list wins
    <coolo> if that one is a workaround needle -> softfail
    <adamw> that seems like undesirably unpredictable behaviour...would you take a fix for it? and which way around would you want it to work?
    <okurz> adamw: IMHO 1. soft fail needles, 2. reverse alphabetical order --> more recent ones with same name but newer timestamp win
    <okurz> a soft-fail needle for something *not* showing up is tricky so I would have rather needled the *presence* of a blank background (with missing default session name). A softfail needle should always be designed to only match whenever the bug is present. Does this make sense to you?
    <okurz> adamw: you can also say it this way: A soft-fail is any fail (regardless of priority or severity) *whenever* you can workaround it

Basically right now when multiple needles match (at the same level, or just match? unclear) the first in alphabetical order is taken, with no consideration as to whether it's a regular or 'workaround' needle. We think that 'workaround' needles should always match ahead of regular needles. As usual I'll work on this if I find time, but no promises :)

**Comment by coolo** [comment]
Adding a consideration based on properties needs to go here:

https://github.com/os-autoinst/os-autoinst/blob/master/ppmclibs/tinycv.pm#L128

Tests are easy to add too.

**Comment by AdamWill** [comment]
https://github.com/os-autoinst/os-autoinst/pull/532
